.github/workflows/run_periodic_tests.yml

```diff
@@ -88,6 +88,7 @@ jobs:
         run: python -m pip install nox
 
       - name: Install SuiteSparse and SUNDIALS on GNU/Linux and macOS
+        timeout-minutes: 10
```

Comment:
The reason for adding a timeout is because of runs like these: https://github.com/pybamm-team/PyBaMM/actions/runs/7809381767/job/21301125955

I have never faced deadlocks like this locally, but I remember that this started happening on GitHub Actions runners after #3646 was resolved. This should be a sufficient workaround since we can't ascertain the cause that easily.